client/src/endpoint/transaction.rs

```diff
@@ -162,6 +162,188 @@ mod all_transactions_test {
     }
 }
 
+/// An endpoint that accesses all the effects that occurred as a result of a transaction
+#[derive(Debug, Clone)]
+pub struct Effects {
+    hash: String,
+    cursor: Option<String>,
+    order: Option<Order>,
+    limit: Option<u32>,
+}
+
+impl Effects {
+    /// Returns a new endpoint for effects. Hand this to the client in order
+    /// to request effects for a specific transaction by hash
+    pub fn new(hash: &str) -> Self {
+        Self {
+            hash: hash.to_string(),
+            cursor: None,
+            order: None,
+            limit: None,
+        }
+    }
+
+    /// Starts the page of results at a given cursor
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// use stellar_client::sync::Client;
```

Comment:
We've just started moving the example that makes queries to horizon to just the struct definition itself. This way we provide a large example for clarity and reduce the number of http calls we make.

Here's an example of the new approach:
https://github.com/kbacha/stellar-sdk/blob/master/client/src/endpoint/account.rs#L456-L558

(note: we haven't converted the old examples yet)

Here's what the "new" way looks like in `cargo doc`:

![image](https://user-images.githubusercontent.com/2043529/38427042-0d322844-396d-11e8-8c67-f39bd7c61105.png)